### PR TITLE
AL-2317  CloudGuard AWP (AWS) - Terraform Module

### PR DESCRIPTION
**Pull request by bzemms**
## CloudGuard AWP (AWS) - Terraform Module
This pull request introduces the implementation of the Terraform CloudGuard(Dome9) AWP for AWS module:

This Terraform module is designed to onboard AWS accounts to CloudGuard(Dome9) AWP (Agentless Workload Posture) service.
(https://www.checkpoint.com/dome9/) 

This module uses [Check Point CloudGuard Dome9 Provider](https://registry.terraform.io/providers/dome9/dome9/latest/docs)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+  policy_arn = aws_iam_policy.CloudGuardAWP.arn
+}
+# end resources for CloudGuardAWPCrossAccountRole
+
+# Cross account role policy
+# The CloudGuardAWPCrossAccountRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy" "CloudGuardAWPCrossAccountRolePolicy_InAccount" {
+  count       = local.scan_mode == "inAccount" ? 1 : 0
+  name        = "CloudGuardAWPCrossAccountRolePolicy_InAccount"
+  description = "Policy for CloudGuard AWP Cross Account Role"
```
Comment (omartawfeeq) [inline main.tf:148]: change from "Policy for CloudGuard AWP Cross Account Role" to "Policy for CloudGuard AWP Cross Account Role - InAccount Mode"

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+          "kms:CreateKey",
+        ]
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRolePolicyAttachment resource attaches the CloudGuardAWPCrossAccountRolePolicy to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy_attachment" "CloudGuardAWPCrossAccountRolePolicyAttachment" {
```
Comment (omartawfeeq) [inline main.tf:222]: change resource name from "CloudGuardAWPCrossAccountRolePolicyAttachment" to "CloudGuardAWPCrossAccountRolePolicyAttachment_InAccount"

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+resource "aws_iam_policy_attachment" "CloudGuardAWPCrossAccountRolePolicyAttachment_SaaS" {
+  count      = local.scan_mode == "saas" ? 1 : 0
+  name       = "CloudGuardAWPCrossAccountRolePolicyAttachment_SaaS"
+  policy_arn = aws_iam_policy.CloudGuardAWPCrossAccountRolePolicy_SaaS[count.index].arn
+  roles      = [aws_iam_role.CloudGuardAWPCrossAccountRole.name]
+}
+# END Cross account role policy
+
+# The CloudGuardAWPSnapshotsUtilsFunctionZip resource defines http data source to download the remote function file from S3 pre-signed URL.
+data "http" "CloudGuardAWPSnapshotsUtilsFunctionZip" {
```
Comment (omartawfeeq) [inline main.tf:239]: this resouce have the name CloudGuardAWPSnapshotsUtilsFunctionZip and the next resource have the same name, maybe change them to be unique names give fast differentiation for them

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+# The CloudGuardAWPSnapshotsUtilsLambdaExecutionRoleAttachment resource attaches the CloudGuardAWPSnapshotsPolicy to the CloudGuardAWPSnapshotsUtilsLambdaExecutionRole.
+resource "aws_iam_role_policy_attachment" "CloudGuardAWPSnapshotsUtilsLambdaExecutionRoleAttachment" {
+  role       = aws_iam_role.CloudGuardAWPSnapshotsUtilsLambdaExecutionRole.name
+  policy_arn = aws_iam_policy.CloudGuardAWPSnapshotsPolicy.arn
+}
+# END AWP proxy lambda function role
+
+# AWP proxy lambda function role policy
+# The CloudGuardAWPLambdaExecutionRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPSnapshotsUtilsFunction.
+resource "aws_iam_policy" "CloudGuardAWPLambdaExecutionRolePolicy" {
```
Comment (omartawfeeq) [inline main.tf:372]: change to "CloudGuardAWPLambdaExecutionRolePolicy_InAccount" to make all resources name conssitent with the naming convention we used

File: versions.tf [hunk 5]
```diff
@@ -0,0 +1,20 @@
+terraform {
+  required_providers {
+    dome9 = {
+      source = "dome9/dome9"
+      # version = "1.29.6" # TBD
```
Comment (omartawfeeq) [inline versions.tf:5]: change to >= 1.29.7

**Review by omartawfeeq — COMMENTED** [review]
have some comments

File: README.md [hunk 6]
```diff
@@ -1,2 +1,135 @@
...
+
+## Usage
+
+```hcl
+module "terraform-dome9-awp-aws" {
+  source = "dome9/awp/aws"
+
+  # The Id of the AWS account,onboarded to CloudGuard (can be either the Dome9 Cloud Account ID or the AWS Account Number)
+  awp_cloud_account_id = dome9_cloudaccount_aws.my_aws_account.id
+
```
[the remark this answers is not in the document]
Comment (omartawfeeq) [reply README.md:24]: i mean when using the module, usually we use the module block as : 
`module "terraform-dome9-awp-aws" {
	source = "github.com/dome9/terraform-dome9-awp-aws"
	version = "1.0.0"`

File: README.md [hunk 7]
```diff
@@ -1,2 +1,135 @@
...
+| <a name="input_awp_scan_mode"></a> [awp_scan_mode](#input\_awp\_scan\_mode) | The scan mode for the AWP `[ "inAccount" \| "saas" ]`| `string` | "inAccount" | yes |
+| <a name="input_awp_cross_account_role_name"></a> [awp_cross_account_role_name](#input\_awp\_cross\_account\_role\_name) | AWP Cross account role name | `string` | `CloudGuardAWPCrossAccountRole` | no |
+| <a name="input_awp_cross_account_role_external_id"></a> [awp_cross_account_role_external_id](#input\_awp\_cross\_account\_role\_external\_id) | AWP Cross account role external id | `string` | `null` (auto-generated) | no |
+| <a name="input_awp_additional_tags"></a> [awp_additional_tags](#input\_awp\_additional\_tags) | Additional tags to be added to the module resources | `map(string)` | `{}` | no |
+|  [awp_account_settings_aws](#input\_awp\_account\_settings\_aws) | AWP Account settings for AWS | object | `null` | no |
+
+<br/>
+
+**<a name="input_awp_account_settings_aws"></a> [awp_account_settings_aws](#input\_awp\_account\_settings\_aws) variable is an object that contains the following attributes:**
```
[the remark this answers is not in the document]
Comment (omartawfeeq) [reply README.md:80]: i opened it, from the browser it is shown correctly , but from the IDE viewer, is not, but it is not critical, resolving

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,668 @@
...
+  tags = local.common_tags
+}
+
+resource "aws_lambda_permission" "allow_cloudguard" {
+  statement_id  = "AllowExecutionFromCloudGuard"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction.function_name
+  principal     = "s3.amazonaws.com"
+  source_arn    = "arn:${data.aws_partition.current.partition}:s3:::${local.agentless_bucket_name}/*"
+}
```
[the remark this answers is not in the document]
Comment (omartawfeeq) [reply main.tf]: after i investigated it, it is useless, we can remove it

**Review by yuval-avram — APPROVED** [review]

**Pull request merged**